plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -152,21 +155,49 @@ func (c *MaxPDVolumeCountChecker) filterVolumes(volumes []api.Volume, namespace
 		} else if vol.PersistentVolumeClaim != nil {
 			pvcName := vol.PersistentVolumeClaim.ClaimName
 			if pvcName == "" {
-				return fmt.Errorf("PersistentVolumeClaim had no name: %q", pvcName)
+				// if the PVC does not have a name, log the error
+				// and count the PV towards the PV limit
+				// generate a random volume ID since its required for de-dup
+				glog.Errorf("PersistentVolumeClaim with no name found in namespace: %q", namespace)
+				source := rand.NewSource(time.Now().UnixNano())
+				generatedID := "noClaimName" + strconv.Itoa(rand.New(source).Intn(1000000))
+				filteredVolumes[generatedID] = true
+				return nil
 			}
 			pvc, err := c.pvcInfo.GetPersistentVolumeClaimInfo(namespace, pvcName)
 			if err != nil {
-				return err
+				// if the PVC is not found, log the error
+				// and count the PV towards the PV limit
+				// generate a random volume ID since its required for de-dup
+				glog.Error(err)
+				source := rand.NewSource(time.Now().UnixNano())
+				generatedID := "missingPVC" + strconv.Itoa(rand.New(source).Intn(1000000))
+				filteredVolumes[generatedID] = true
+				return nil
```

Comment:
This is the only scenario, I believe, that is valid -- the PV is currently attached but the PVC has been deleted.